core/src/main/java/com/linecorp/armeria/client/ClientRequestContextBuilder.java

```diff
@@ -103,29 +116,35 @@ public ClientRequestContextBuilder requestOptions(RequestOptions requestOptions)
      * Returns a new {@link ClientRequestContext} created with the properties of this builder.
      */
     public ClientRequestContext build() {
-        final Endpoint endpoint;
-        if (this.endpoint != null) {
-            endpoint = this.endpoint;
+        final EndpointGroup endpointGroup;
+        if (this.endpointGroup != null) {
+            endpointGroup = this.endpointGroup;
         } else {
-            endpoint = Endpoint.parse(authority());
+            endpointGroup = Endpoint.parse(authority());
         }
-
-        final CancellationScheduler responseCancellationScheduler;
+        final DefaultClientRequestContext ctx;
         if (timedOut()) {
-            responseCancellationScheduler = CancellationScheduler.finished(false);
+            ctx = new DefaultClientRequestContext(
+                    eventLoop(), meterRegistry(), sessionProtocol(), id(), method(), requestTarget(), options,
+                    request(), rpcRequest(), requestOptions, CancellationScheduler.finished(false),
+                    isRequestStartTimeSet() ? requestStartTimeNanos() : System.nanoTime(),
+                    isRequestStartTimeSet() ? requestStartTimeMicros() : SystemInfo.currentTimeMicros());
```

Comment:
Should we remove L133-L136 and deduplicate code?
For example:
```java
if (timedOut) {
    responseCancellationScheduler = CancellationScheduler.finished(false);
} else {
    responseCancellationScheduler = CancellationScheduler.ofClient(0);
}
final DefaultClientRequestContext ctx = new DefaultClientRequestContext(
        eventLoop(), meterRegistry(), sessionProtocol(), id(), method(), requestTarget(), options,
        request(), rpcRequest(), requestOptions, responseCancellationScheduler,
        isRequestStartTimeSet() ? requestStartTimeNanos() : System.nanoTime(),
        isRequestStartTimeSet() ? requestStartTimeMicros() : SystemInfo.currentTimeMicros());
if (!timedOut) {
    ctx.whenInitialized().handle((unused1, unused2) -> {
        ctx.responseCancellationScheduler().initAndStart(ctx.eventLoop(), noopCancellationTask);
        return null;
    });
}
```

Would you add test cases to ensure a `ClientRequestContext` is correctly created with a given `EndpointGroup`?